src/components/marker/timeline-preview.tsx

```diff
@@ -0,0 +1,132 @@
+import {h, Fragment, VNode, Component} from 'preact';
+import * as styles from './timeline-preview.scss';
+import {icons} from '../icons';
+import {ItemTypes, ThumbnailInfo} from "../../types/timelineTypes";
+import {CuePointMarker} from "../../../flow-typed/types/cue-point-option";
+
+const {Icon} = KalturaPlayer.ui.components;
+const {withText, Text} = KalturaPlayer.ui.preacti18n;
+
+const translates = {
+  questionTranslate: <Text id="timeline.question_title">Question</Text>,
+  reflectionPointTranslate: <Text id="timeline.reflection_point_title">Reflection Point</Text>,
+  hotspotTranslate: <Text id="timeline.hotspot_title">Hotspot</Text>,
+  aoaTranslate: <Text id="timeline.audience_asked_title">Audience Asked</Text>
+};
+
+interface TimelinePreviewProps {
+  onArrowClick: () => void,
+  cuePointsData: Array<CuePointMarker>,
+  thumbnailInfo: any,
+  isNavigationPluginOpen: () => boolean,
+  shouldRenderArrowButton: () => boolean,
+  chaptersData: any[],
+  questionTranslate?: string,
+  reflectionPointTranslate?: string,
+  hotspotTranslate?: string,
+  aoaTranslate?: string
+}
+
+const getFramePreviewImgContainerStyle = (thumbnailInfo: ThumbnailInfo) => {
+  return thumbnailInfo ? {
+    height: `${thumbnailInfo.height}px`,
+    width: `${thumbnailInfo.width}px`
+  } : {};
+};
+const getFramePreviewImgStyle = (thumbnailInfo: ThumbnailInfo) => {
+  if (thumbnailInfo) {
+    let framePreviewImgStyle = `height: 100%; width: 100%; background: url(${thumbnailInfo.url});`;
+    framePreviewImgStyle += `background-position: -${thumbnailInfo.x}px -${thumbnailInfo.y}px;`;
+    return framePreviewImgStyle;
+  }
+  return '';
+};
+
+interface TitleProps {
+  id: string,
+  iconPath: string,
+  children: VNode | string
+}
+
+const Title = (({id, iconPath, children}: TitleProps) => {
+  return (
+  <div className={styles.titleWrapper}>
+    <Icon
+      id={id}
+      height={icons.SmallSize}
+      width={icons.SmallSize}
+      viewBox={`0 0 ${icons.SmallSize} ${icons.SmallSize}`}
+      path={iconPath}/>
+    <span className={styles.title}>{children}</span>
+  </div>
+  );
+});
+
+@withText(translates)
+export class TimelinePreview extends Component<TimelinePreviewProps> {
+  private _handleClick = () => {
+    this.props.onArrowClick();
+  };
+
+  private _renderHeader() {
+    // organize the header by cuePoint types
+    const hotspots = this.props.cuePointsData.filter(cuePoint => cuePoint.type === ItemTypes.Hotspot);
+    const quizQuestions = this.props.cuePointsData.filter(cuePoint => cuePoint.type === ItemTypes.QuizQuestion);
+    const chapters = this.props.chaptersData;
+    const answerOnAir = this.props.cuePointsData.filter(cuePoint => cuePoint.type === ItemTypes.AnswerOnAir);
```

Comment:
I believe better make:
`
const initialData =  {
[ItemTypes.QuizQuestion]: [],
[ItemTypes.Hotspot]: [],
[ItemTypes.AnswerOnAir]: []
};
this.props.cuePointsData.reduce((acc, next) => { ...acc,  [cuePoint.type]: [...acc[cuePoint.type],  cuePoint]}, initialData);`
`reduce` iterates array of `cuePointsData` only once instead of 3 iterations with `filter`